src/Services/MetaWeblogService.cs

```diff
@@ -160,7 +160,7 @@ public int AddCategory(string key, string username, string password, NewCategory
 
         private void ValidateUser(string username, string password)
         {
-            if (_userServices.ValidateUser(username, password))
+            if (_userServices.ValidateUser(username, password)==false)
```

Comment:
!_userServices.ValidateUser(…) instead of == false would be more appropriate and following current codestyle.